### PR TITLE
Bugfix/zcs 1098

### PR DESCRIPTION
**Pull request by shri314**
Please see - https://jira.corp.synacor.com/browse/ZCS-1098.

This is only for review, do not merge, this is still work in progress

### CODE REVIEW

**Comment by shri314** [comment]
Incorporated all review comments (except plobbes comment w.r.to https)

**Comment by shri314** [comment]
@plobbes: I have incorporated all your comments except https vs http, as that is out of scope of the current change.

File: Makefile [hunk 1]
```diff
@@ -46,6 +58,45 @@ zimbra-chat-pkg: require-pkg-release
...
+
+########################################################################################################
+
+gen-hash-track:
+	mkdir -p hash-track/
+	sha512sum $(TRACK_IN) > hash-track/$(TRACK_OUT).hash
+	@if [ "$$(git status -s hash-track/$(TRACK_OUT).hash)" != "" ]; \
+	then \
+	   echo; \
+           echo "ERROR: ---------------------------------------------------------------------"; \
```
Comment (plobbes) [inline Makefile]: nit: inconsistent indentation here (and 2 lines below)

Comment (shri314) [inline Makefile]: Makefile uses tabs. Its some display issue with github :-)

**Pull request merged**